packages/block-library/src/image/style.scss

```diff
@@ -44,7 +44,7 @@
 
 	&.alignfull a,
 	&.alignwide a {
-		width: 100%;
+		width: auto;
```

Comment:
I believe this change needs to be limited to links inside captions, and not the linked image.

Reply:
Hi @carolinan, Thanks for pointing this out.

I have addressed the feedback and now specifically targeting the caption width.